pkg/api/groups.go

```diff
@@ -71,25 +114,61 @@ func (a *API) GetGroup(w http.ResponseWriter, r *http.Request, groupId string) {
 
 // Create Group
 // (POST /api/v1/admin/groups)
+// Creates a group in cognito if cognito is the selected idp, otherwise will create an Internal Granted group
 func (a *API) CreateGroup(w http.ResponseWriter, r *http.Request) {
 	ctx := r.Context()
-	if a.Cognito == nil {
-		apio.ErrorString(ctx, w, "api not available", http.StatusBadRequest)
-		return
-	}
 	var createGroupRequest types.CreateGroupJSONRequestBody
 	err := apio.DecodeJSONBody(w, r, &createGroupRequest)
 	if err != nil {
 		apio.Error(ctx, w, apio.NewRequestError(err, http.StatusBadRequest))
 		return
 	}
-	group, err := a.Cognito.CreateGroup(ctx, cognitosvc.CreateGroupOpts{
+	// if a.Cognito == nil {
+	// 	//create cognito group
+
+	// 	group, err := a.Cognito.CreateGroup(ctx, cognitosvc.CreateGroupOpts{
+	// 		Name:        createGroupRequest.Name,
+	// 		Description: aws.ToString(createGroupRequest.Description),
+	// 	})
+	// 	if err != nil {
+	// 		apio.Error(ctx, w, err)
+	// 		return
+	// 	}
+	// 	apio.JSON(ctx, w, group.ToAPI(), http.StatusCreated)
+	// } else {
+	// Create internal group
+	group := identity.Group{
+		ID:          createGroupRequest.Name,
+		IdpID:       createGroupRequest.Name,
 		Name:        createGroupRequest.Name,
-		Description: aws.ToString(createGroupRequest.Description),
-	})
+		Description: *createGroupRequest.Description,
+		Status:      types.IdpStatusACTIVE,
+		Source:      types.INTERNAL,
+		Users:       *createGroupRequest.Members,
```

Comment:
If members is not optional and should instead be an empty list. Update the API schema so that this is not a pointer to a slice.